lib/action_policy/lookup_chain.rb

```diff
@@ -24,14 +24,27 @@ class NamespaceCache
       class << self
         attr_reader :store
 
-        def fetch(namespace, policy)
+        def put_if_absent(scope, namespace, policy)
+          local_store = store[scope][namespace]
+          return local_store[policy] if local_store.key?(policy)
+          local_store[policy] ||= yield
+        end
+
+        def fetch(namespace, policy, strict:, &block)
```

Comment:
It would be great to have some tests for this change. Could you please do that?

Reply:
done. feel free to take over this PR to resolve those tests (if anything is missing).